pilot/pkg/bootstrap/server.go

```diff
@@ -883,6 +883,7 @@ func (s *Server) initServiceControllers(args *PilotArgs) error {
 		switch serviceRegistry {
 		case serviceregistry.MockRegistry:
 			s.initMemoryRegistry(serviceControllers)
+			addMockService(serviceControllers)
```

Comment:
why don't you add it within the `initMemoryRegistry` where we add all the mock stuff?

Reply:
```
why don't you add it within the initMemoryRegistry where we add all the mock stuff?
```
I am not sure if initMemoryRegistry is just for testing purposes.

In fact, "istio" can use ConfigRegistry to simulate eds, so I closed this pr.